.github/workflows/ci.yaml

```diff
@@ -0,0 +1,73 @@
+name: CI
+
+on: [pull_request, push]
+
+jobs:
+  lint:
+    name: Lint with `mix check`
+    runs-on: ubuntu-22.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Install OTP and Elixir
+      uses: erlef/setup-beam@v1
+      id: beam
+      with:
+        version-file: .tool-versions
+        version-type: strict
+
+    - name: Cache deps and build
+      uses: actions/cache@v3
+      with:
+        path: |
+          deps
+          _build
+        key: "${{ runner.os }}-${{ steps.beam.outputs.otp-version }}-${{ steps.beam.outputs.elixir-version }}-${{ hashFiles('**/mix.lock') }}"
```

Comment:
as The other PR, to avoid an invalid cache between lint and test CI, prefix this with ```lint-``` and ```test-```

Reply:
Yes, it will be rebased once #9 is merged